### PR TITLE
Trista's and Sofia's chatbot, w2

### PR DESCRIPTION
**Pull request by sofia32057**
## Netlify link
[tristas-and-sofias-chatbot.netlify.app](https://tristas-and-sofias-chatbot.netlify.app/)

## Collaborators
[trista1987]

### CODE REVIEW

File: code/script.js [hunk 1]
```diff
@@ -1,12 +1,63 @@
+// Global variables
+let userName = ""
+let mainFoodChoice = ""
+let foodSubtype = ""
+let partySize = "10"
+let orderValue 
+let onePizzaPrice = 15
+let oneSaladPrice = 18
+let onePastaPrice = 12
+let confirmPrice = ""
```
Comment (HIPPIEKICK) [inline code/script.js]: Maybe this could be of type boolean? 👀

File: code/script.js [hunk 2]
```diff
@@ -1,12 +1,63 @@
...
+
+// ------------------------------------------------
 // DOM selectors (variables that point to selected DOM elements) goes here 👇
 const chat = document.getElementById('chat')
+const inputWrapper = document.getElementById("input-wrapper") // Select the input wrapper
```
Comment (HIPPIEKICK) [inline code/script.js]: Make sure to be consistent with whether you're using single or double quotes

File: code/script.js [hunk 3]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
 }
 
-// Eventlisteners goes here 👇
+// function to save name and send to showMessage, and move on
+const submitName = (event) => {
+  event.preventDefault() 
+  userName = nameInput.value
+  nameInput.value = ""
+  showMessage(userName, "user")
+  setTimeout(() => chooseFood(), 1000) // Go to next step, chooseFood
```
Comment (HIPPIEKICK) [inline code/script.js]: If you don't pass any params, you could just write like this:
`setTimeout(chooseFood, 1000)`

File: code/script.js [hunk 4]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
+}
+
+// function to ask for food choice and send to displayFood
+const chooseFood = () => {
+  showMessage(`What kind of food would you like us to serve, ${userName}?`, "bot")
+  displayFood();
```
Comment (HIPPIEKICK) [inline code/script.js:102]: Are you using semi-colons or not? 😉 Make up your minds

File: code/script.js [hunk 5]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
+  partyInput.value = "10"
+  partyInput.max = "50"
+  partyLabel.textContent = `Party size: ${partySize}`
+  partyLabel.htmlFor = "party-input"
+  partyButton.type = "submit"
+  partyButton.textContent = "Send"
+
+  partyForm.append(partyLabel, partyInput, partyButton)
+  inputWrapper.replaceChild(partyForm, subtypeDiv)
+}
```
Comment (HIPPIEKICK) [inline code/script.js]: I love that you chose a range slider for this question! You need to validate the input here though, it doesn't make sense to be able to order 0 pizzas.

File: code/script.js [hunk 6]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
+  showMessage(partySize, "user")
+  calculateOrderValue()
+}
+
+const calculateOrderValue = () => {
+  inputWrapper.removeChild(partyForm)
+  if (mainFoodChoice === "Pizza"){
+    orderValue = partySize * onePizzaPrice
+  } else if (mainFoodChoice === "Pasta"){
+    orderValue = partySize *onePastaPrice
```
Comment (HIPPIEKICK) [inline code/script.js]: Tiny thing, but it's the tiny things that adds up to make the code look clean - there should be a space between the * and the variable 🔍

File: code/script.js [hunk 7]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
+  }else {
+    orderValue = partySize * oneSaladPrice
+  }
+  showMessage(`You have ordered ${partySize} ${foodSubtype}, here is your bill: ${orderValue} €. Is that ok?`, "bot")
+  setTimeout(userConfirm, 1000)
+}
+
+const userConfirm = () => {
```
Comment (HIPPIEKICK) [inline code/script.js]: Could this function be called something like confirmPurchase or confirmOrder?

File: code/script.js [hunk 8]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
+    showMessage(`Oh I am sorry for that! Let's see you again!`, "bot")
+    inputWrapper.innerHTML=""
+  })
+  setTimeout(restartChat, 7000)
+}
+
+// Reload the chat
+const restartChat = () => {
+  location.reload()
+}
```
Comment (HIPPIEKICK) [inline code/script.js]: Nice!

File: code/script.js [hunk 9]
```diff
@@ -29,25 +78,213 @@ const showMessage = (message, sender) => {
...
+// Party
+partyInput.addEventListener("input", () => {
+  partySize = partyInput.value // Update variable partySize
+  partyLabel.textContent = `Party size: ${partySize}` // Update input label
+})
+partyButton.addEventListener("click", submitPartysize)
+
+
+
+
```
Comment (HIPPIEKICK) [inline code/script.js]: It's enough to end with one empty line 😊

**Review by HIPPIEKICK — CHANGES_REQUESTED** [review]
Good job with your chatbot Trista & Sofia! I like that you made the project your own by using the createElement method and using some fun input types (range slider 🥳). Overall your code is organized and clean, but I would like to ask you to clean it up just a tiny bit by:
- being consistent with whether you're using semi-colons or not
- being consistent with whether you're using single or double quotes

and apart from that - fix the little bug that allows the user to order 0 pizzas, and then you're good to go.

Feel free to reach out on Stack Overflow if you have any doubts about how to approach it!

**Review by AntonellaMorittu — APPROVED** [review]

File: code/script.js (hunk 1 above)
Comment (trista1987) [reply code/script.js]: if change it to be boolean. Shoud I write `let onePizzaPrice = true` ? I'm a little confused. Could you please explain more? 😃

**Pull request closed without merging**